### PR TITLE
Adding a 'not' filter

### PR DESCRIPTION
**Pull request by scotwk**
@kapilt - is this what you're going for, or is there more to this filter than this?

I've tested it and it works in a basic test, but my unit test doesn't work because the tests don't utilize a query manager, so I have to figure out how to write a `__call__` method (or modify the tests to use a query manager).

### CODE REVIEW

File: c7n/filters/core.py [hunk 1]
```diff
@@ -200,6 +203,27 @@ def process(self, resources, events=None):
...
 
+class Not(Filter):
+    
+    def __init__(self, data, registry, manager):
+        super(Not, self).__init__(data)
+        self.registry = registry
+        self.filters = registry.parse(self.data.values()[0], manager)
+        self.manager = manager
+
+    def process(self, resources, events=None):
```
Comment (kapilt) [inline c7n/filters/core.py]: event is singular at the moment

**Comment by kapilt** [comment]
yup, this was the goal, thanks!

File: c7n/filters/core.py (hunk 1 above)
Comment (scotwk) [inline c7n/filters/core.py]: @kapilt - I just copied from the And filter.  Should that one be singular also?

**Comment by scotwk** [comment]
@kapilt - I got all the tests passing, but I'm trying to figure out if I need to special case `not` in filters/vpc.py and filters/related.py.  Also, I should probably add a test that hits the `process_set function in the new `Not` filter for coverage purposes.

**Comment by scotwk** [comment]
@kapilt - I added a test to complete coverage of the new code.  Looks like the `and` and `or` in filters/vpc.py and filters/related.py are independent of what I added.

Should be good to go now.

**Pull request merged**